### PR TITLE
Feature/login ui basic

### PR DESCRIPTION
**Pull request by tomklecha**
basic UI concept
@Canato I have a struggle with ServiceLocator class - i dont know how to properly create the class, so all will allow me to attach the locator in the fragment.

### CODE REVIEW

**Comment by Canato** [comment]
> i dont know how to properly create the class, so all will allow me to attach the locator in the fragment.

Did you take a look on the questions ServiceLocator? @tomklecha

File: app/build.gradle [hunk 1]
```diff
@@ -42,4 +42,5 @@ dependencies {
     implementation project(':common')
     implementation project(':features:questions')
     implementation project(':features:login')
+    implementation 'com.google.android.material:material:1.1.0'
```
Comment (Canato) [inline app/build.gradle:45]: Why you need this library?

Comment (tomklecha) [reply app/build.gradle:45]: i add a button as a on click in main activity - can be changed

File: features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    fun getPresenter() : LoginContract.Presenter =
+        LoginPresenter(
+            getInteractor()
+        )
+
+}
+
+private fun getInteractor(): LoginContract.Interactor = LoginInteractor(getRepository())
+private fun getRepository(): LoginContract.Repository = LoginRepository(getApi())
+private fun getApi(): LoginFirebaseApi = LoginFirebaseApiDefault(TODO())
```
Comment (tomklecha) [inline features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt:23]: @Canato in this moment i had problem, how to properly implement getApi.
This one is not done yet on Questions module

File: app/build.gradle (hunk 1 above)
Comment (tomklecha) [reply app/build.gradle:45]: button changed from floating button to normal button -> implementation deleted

File: features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt (hunk 2 above)
Comment (tomklecha) [reply features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt:23]: @Canato our whatsupp conversation

**Comment by tomklecha** [comment]
@Canato have a look please

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -2,8 +2,10 @@
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/activity_main"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:backgroundTint="#FFFFFF"
```
Comment (Canato) [inline app/src/main/res/layout/activity_main.xml:8]: Let's create this color on `@color/white`

File: features/login/src/main/java/com/can_apps/login/bresenter/LoginPresenter.kt [hunk 4]
```diff
@@ -33,7 +34,9 @@ internal class LoginPresenter(
         if (isLoginValid && isPasswordValid) {
             when (interactor.loginUser(loginNameDomain, passwordDomain)) {
                 LoginDomain.Success -> view.showSuccess()
-                LoginDomain.Fail(LoginErrorDomain("flipflops")) -> view.showError("flipflops")
+                LoginDomain.Fail(LoginErrorDomain("Sorry, login is not available")) -> view.showError(
+                    "Sorry, login is not available"
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/bresenter/LoginPresenter.kt:38]: We need to talk about this. Remember on Monday please

File: features/login/src/test/java/com/can_apps/login/bresenter/LoginPresenterTest.kt [hunk 5]
```diff
@@ -79,7 +79,7 @@ internal class LoginPresenterTest {
 
         val password = "pass"
         val loginName = "loginName"
-        val error = "flipflops"
+        val error = "Sorry, login is not available"
```
Comment (Canato) [inline features/login/src/test/java/com/can_apps/login/bresenter/LoginPresenterTest.kt:82]: We should not change this test. That's the idea of the test. Let's talk on Monday

File: features/login/src/test/java/com/can_apps/login/bresenter/LoginPresenterTest.kt [hunk 6]
```diff
@@ -93,7 +93,7 @@ internal class LoginPresenterTest {
 
         //THEN
         verify {
-            view.showError(error)
+            view.showError(any())
```
Comment (Canato) [inline features/login/src/test/java/com/can_apps/login/bresenter/LoginPresenterTest.kt:96]: We should not change this test. That's the idea of the test. Let's talk on Monday

File: features/login/src/main/res/layout/fragment_login.xml [hunk 7]
```diff
@@ -5,9 +5,10 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:layout_margin="8dp"
+    android:background="#FFFFFF"
     android:orientation="vertical">
 
-    <com.google.android.material.textfield.TextInputLayout
+    <EditText
```
Comment (Canato) [inline features/login/src/main/res/layout/fragment_login.xml:11]: https://github.com/Canato/Purity-Test/pull/2#discussion_r403488041 

What happened? Why are we undoing ?

File: features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    private val auth: FirebaseAuth = FirebaseAuth.getInstance()
+
+    fun getPresenter(): LoginContract.Presenter =
+        LoginPresenter(
+            getInteractor()
+        )
+
+    private fun getInteractor(): LoginContract.Interactor = LoginInteractor(getRepository())
+    private fun getRepository(): LoginContract.Repository = LoginRepository(getApi())
+    private fun getApi(): LoginFirebaseApi = LoginFirebaseApiDefault(auth)
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt:23]: What is not working here? @tomklecha

File: features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+import com.can_apps.login.core.LoginContract
+import com.can_apps.login.core.LoginInteractor
+import com.can_apps.login.data.LoginFirebaseApi
+import com.can_apps.login.data.LoginFirebaseApiDefault
+import com.can_apps.login.data.LoginRepository
+import com.google.firebase.auth.FirebaseAuth
+
+internal class LoginServiceLocator(private val context: Context) {
+
+    private val auth: FirebaseAuth = FirebaseAuth.getInstance()
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt:14]: Do we need `auth` as variable? or it is used only in one place?

Comment (tomklecha) [reply features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt:14]: if i wont give to ServiceLocator the auth variable, then i cannot use function
 private fun getApi(): LoginFirebaseApi = LoginFirebaseApiDefault(auth)
so thanks to that i can implement the presenter

File: features/login/src/main/res/layout/fragment_login.xml (hunk 7 above)
Comment (tomklecha) [reply features/login/src/main/res/layout/fragment_login.xml:11]: To do the full test, i needed two edit texts to put login and password - to prove if everything is working as it should. So far yes

File: features/login/src/test/java/com/can_apps/login/bresenter/LoginPresenterTest.kt (hunk 6 above)
Comment (tomklecha) [reply features/login/src/test/java/com/can_apps/login/bresenter/LoginPresenterTest.kt:96]: will change it

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (tomklecha) [reply app/src/main/res/layout/activity_main.xml:8]: i willl change it now

File: features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt (hunk 8 above)
Comment (tomklecha) [reply features/login/src/main/java/com/can_apps/login/app/LoginServiceLocator.kt:23]: it was not working before - since adding auth variable now its fine

**Pull request merged**